internal/util/url.go

```diff
@@ -75,11 +76,10 @@ func (f ReadUnmounter) Close() error {
 }
 
 // FetchResourceAsReader returns a ReadCloser to the data at the url specified. The caller is responsible for
-// closing the reader.
-func FetchResourceAsReader(l *log.Logger, u url.URL) (io.ReadCloser, error) {
+// closing the reader. client is only necessary if the url is http(s).
+func FetchResourceAsReader(l *log.Logger, client HttpClient, u url.URL) (io.ReadCloser, error) {
```

Comment:
> I'm a bit bothered by the passing around of the HttpClient everywhere, then we arrive here to learn that the HttpClient may not even be applicable for non-http urls.

Unless we want to introduce globals the timeout values will need to passed around everywhere that might make an http call in some form or another, whether that be wrapped in an `HttpClient` or as a `types.Timeouts`. One (minor) advantage of using the clients is that we can reuse connections automatically by sharing the client.
The other problem is that url is not specific to http(s) as it can be used for oem:// or data:// schemes. This means if we attach the timeout values (whether that be via `HttpClient`, `types.Timeouts`, etc) to the url, they might be unused, just as the client parameter might be unused. On the other hand it may be better to abstract that away into the url so the function calls remain a little cleaner without the possibility of the unused parameter.
> it might be easier if you call the function with the logger, url, and a timeout structure (I think we already do this in some places).

We don't; `types.Timeouts` is only used when creating the new client. I don't see a huge advantage to this either since the timeouts might not be used, just like the client, and then we don't even gain reusing connections.